security/pkg/nodeagent/vm/nafactory.go

```diff
@@ -48,10 +49,16 @@ func NewNodeAgent(cfg *Config) (NodeAgent, error) {
 	if err != nil {
 		return nil, err
 	}
-	grpcConn, err := protocol.NewGrpcConnection(cfg.CAClientConfig.CAAddress, dialOpts)
-	if err != nil {
-		return nil, err
+
+	if cfg.CAClientConfig.IsCitadel {
```

Comment:
The right way to handle this if/else is:

```go
if usingNewProtocol {
  grpcConnection(caAddress)
}
else {
  OldProtocol
}
```

We should distinguish based on gRPC API, not the backend implementation. The Citadel/Google CA is just a matter of the connection address argument.